py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,339 @@
+            1: 'HOME',
+            2: 'SEARCHING',
+            3: 'DENIED',
+            4: 'UNKNOWN',
+            5: 'ROAMING'
+        }
+
+        self.bus = SystemBus()
+
+    def wwan_status_nm(self):
+        data = {}
+        # if there is no modem then disply STRING_NO_MODEM
+        full_text = self.py3.safe_format(
+            self.format_no_service, dict(status=STRING_NO_MODEM))
+        color = self.py3.COLOR_BAD
+
+        try:
+            modemmanager_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS)
+            modems = modemmanager_proxy.GetManagedObjects()
+        except:
+            full_text = self.py3.safe_format(self.format_no_service, data)
+            modems = {}
+
+        # browse modems objects
+        for objects in modems.items():
+            modem_path = objects[0]
+            modem_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, modem_path)
+
+            # we can maybe choose another selector
+            eqid = str(modem_proxy.EquipmentIdentifier)
+
+            # use selected modem or first find
+            if self.modem is None or self.modem == eqid:
+
+                # get status informations
+                try:
+                    status = modem_proxy.GetStatus()
+                except:
+                    # if no modem is selected, try all available
+                    continue
+
+                # start to build return data dict
+                try:
+                    data['signal'] = status['signal-quality'][0]
+                    self.py3.threshold_get_color(data['signal'], 'signal')
+                except:
+                    data['signal'] = STRING_UNKNOWN
+
+                try:
+                    highest_access_bit = 1 << (
+                        status['access-technologies'].bit_length() - 1)
+                    data['netgen'] = self.speed[highest_access_bit]
+                    self.py3.threshold_get_color(highest_access_bit, 'netgen')
+                except:
+                    data['netgen'] = STRING_UNKNOWN
+
+                # if registred on network, get operator name
+                try:
+                    if self.rstates[status['m3gpp-registration-state']] in (
+                            'HOME', 'ROAMING'):
+                        data['operator'] = status['m3gpp-operator-name']
+                    else:
+                        data['operator'] = STRING_UNKNOWN
+                except:
+                    data['operator'] = STRING_UNKNOWN
+
+                # use human readable format
+                data['status'] = self.states[status['state']]
+
+                # get status color
+                self.py3.threshold_get_color(status['state'], 'status')
+                self.py3.threshold_get_color(status['state'], 'color')
+
+                # TODO: maybe change to a rstates condition?
+                if status['state'] >= 0 and status['state'] <= 7:  # no service
+                    full_text = self.py3.safe_format(self.format_no_service,
```

Comment:
Checking `state` number seems not need here to me. We try `bearer`, `stats`, etc anyway. Get back nothing. Pass in nothing. Try to remove `format_no_service` format.